etcdserver/v3_server.go

```diff
@@ -81,15 +82,20 @@ func (s *EtcdServer) Range(ctx context.Context, r *pb.RangeRequest) (*pb.RangeRe
 	var err error
 
 	if r.Serializable {
-		var user string
-		user, err = s.usernameFromCtx(ctx)
+		user, rev, err := s.authInfoFromCtx(ctx)
 		if err != nil {
 			return nil, err
 		}
 		result = s.applyV3.Apply(
 			&pb.InternalRaftRequest{
-				Header: &pb.RequestHeader{Username: user},
+				Header: &pb.RequestHeader{Username: user, Revision: rev},
 				Range:  r})
+
+		if rev < s.authStore.Revision() {
```

Comment:
this can just loop until `rev == s.authStore.Revision` instead of bailing out if there was an auth update